Fix Nexmo support 4.0.x

ftreguer opened the pull request:
Fix Nexmo SMS support

Nexmo client needs private key if applicationId is set. DisplayBeforePasswordlessAuthenticationAction uses email configuration for sms notifications.

[x] Remove applicationId default value from NexmoProperties
[x] Add privateKeyContentsPath to NexmoProperties
[x] Use SMS configuration for sms notifications (DisplayBeforePasswordlessAuthenticationAction)
[x] Use default applicationId "CAS" if a private key path is set and applicationId not

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/apereo/cas?pullRequest=3839) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/apereo/cas?pullRequest=3839) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/apereo/cas?pullRequest=3839) it.</sub>

mmoayyed commented on api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/sms/NexmoProperties.java:
```diff
@@ -35,5 +35,10 @@
+    /**
+     * Private key content path to sign Nexmo messages
+     */
+    private String privateKeyContentsPath;
```
This doesn't make much sense. Are we expected to put in contents of something, or is this a path to something/file? It should either be `private String privateKeyContentsPath;` or ` private String privateKeyPath;`. If it's a path, it's best to define it as a Resource, and please document the setting and explain when and how it might be used in combination of other settings.

ftreguer replied:
My bad, private key is for phone call. I'll add secret signature for sms

mmoayyed commented on support/cas-server-support-sms-nexmo/src/main/java/org/apereo/cas/support/sms/NexmoSmsSender.java:
```diff
@@ -24,11 +26,25 @@
+
+        if (StringUtils.isNotBlank(nexmo.getPrivateKeyContentsPath())) {
+            try {
+                // NexmoClient needs privateKeyContents if an applicationId is set
```
Remove line; move into the documentation for the field instead.

mmoayyed commented on support/cas-server-support-sms-nexmo/src/main/java/org/apereo/cas/support/sms/NexmoSmsSender.java:
```diff
@@ -24,11 +26,25 @@
 
     private final NexmoClient nexmoClient;
 
+    private final String DEFAULT_APPLICATION_ID = "CAS";
```
Should be static.

mmoayyed commented on support/cas-server-support-sms-nexmo/src/main/java/org/apereo/cas/support/sms/NexmoSmsSender.java:
```diff
@@ -24,11 +26,25 @@
+                        ? nexmo.getApplicationId() : DEFAULT_APPLICATION_ID)
+                        .privateKeyPath(nexmo.getPrivateKeyContentsPath());
+            } catch (IOException e) {
+                LOGGER.error("No NEXMO private key found", e);
```
`Nexmo`.

ftreguer commented:
So, I've removed applicationId, it's only used for JWT token generation for phone call with private key. Added signatureSecret for signed sms and updated documentation.

mmoayyed commented:
Thank you very much for the contribution. Keep 'em coming!
